src/cljs_test_runner/main.clj

```diff
@@ -44,21 +44,24 @@
         src-path (str/join "/" [src "runner.cljs"])
         out-path (str/join "/" [out "test-runner.js"])
         {:keys [target doo-env]} (case env
-                                   :node {:target :nodejs
-                                          :doo-env :node}
-                                   :phantom {:target :browser
+                                   :node    {:target  :nodejs
+                                             :doo-env :node}
+                                   :phantom {:target  :browser
                                              :doo-env :phantom})]
     (spit src-path test-runner-cljs)
     (try
-      (let [doo-opts {}
-            compiler-opts {:output-to out-path
-                           :output-dir out
-                           :target target
-                           :main 'test.runner
-                           :optimizations :none}]
-        (cljs/build src compiler-opts)
-        (let [{:keys [exit]} (doo/run-script doo-env compiler-opts doo-opts)]
-          (reset! exit-code exit)))
+      (let [doo-opts     {}
+            build-opts   {:output-to     out-path
+                          :output-dir    out
+                          :target        target
+                          :main          'test.runner
+                          :optimizations :none}
+            run-tests-fn #(doo/run-script doo-env build-opts doo-opts)
+            watch-opts   (assoc build-opts :watch-fn run-tests-fn)]
+        (if (seq watch)
+          (cljs/watch (apply cljs/inputs watch) watch-opts)
+          (do (cljs/build src build-opts)
+              (->> (run-tests-fn) :exit (reset! exit-code)))))
       (catch Exception e
         (println e))
       (finally
```

Comment:
This finally block is what cleans up the `test/runner.cljs` file, when you watch there's no way it'll get run. Not sure how best to address this part, ideally it'll be removed after every watch is fired. Since the only way out of the watch is to Ctrl-C the process it'll never get executed.